code/modules/mob/mob.dm

```diff
@@ -42,108 +42,100 @@ var/next_mob_id = 0
 	for(var/datum/gas/trace_gas in environment.trace_gases)
 		t+= "<span class='notice'>[trace_gas.type]: [trace_gas.moles] \n</span>"
 
-	usr.show_message(t, 1)
+	usr << t
 
-/mob/proc/show_message(msg, type, alt, alt_type)//Message, type of message (1 or 2), alternative message, alt message type (1 or 2)
+/mob/proc/show_message(msg, type, alt_msg, alt_type)//Message, type of message (1 or 2), alternative message, alt message type (1 or 2)
 
-	if(!client)	return
+	if(!client)
+		return
 
 	msg = copytext(msg, 1, MAX_MESSAGE_LEN)
 
-	if (type)
-		if(type & 1 && (disabilities & BLIND || paralysis) )//Vision related
-			if (!( alt ))
+	if(type)
+		if(type & 1 && eye_blind )//Vision related
```

Comment:
eye_blind already covers all cases that "(disabilities & BLIND || paralysis)" and some other cases like being flashbanged or wearing welding helmet+sunglasses.